### DISCUSSION TITLE
Is there a way to convert the 3D city model I created in CityEngine to cityjson?

### DISCUSSION THREAD

**Discussion by rhagnus** [cityjson/specs#156] [Q&A] [answered]
Hello, I combined and modeled architectural data, infrastructure data and GIS data of a university campus in CityEngine. I also updated their attribute information. But I want to convert this data to cityjson and fix the deficiencies there. I do this in terms of strengthening the national data infrastructure. CityEngine exports this model as IFC data but it does not export cityjson or citygml. Therefore, if there is a way I can do this, I would be very grateful if you could share it with me.

**Comment by hugoledoux** [comment, marked as answer]
hmmm, never used CityEngine, but indeed it seems CityJSON/GML are not in the list: https://doc.arcgis.com/en/cityengine/latest/help/help-export-models-overview.htm

I guess your best bet would be to export to glTF (since it says it support instances, that is each building/tree is a separate objects), and then write a script to convert to CityJSON or try your luck with FME.

**Reply by rhagnus** [reply]
Thank you for the answer. As you said, I extracted the data as glTF and then converted it to cityjson with Autoconverter software, but only geometry information came. There was no semantic information in the data. I want a questionable result, so it is important for me to transfer semantic information. Can you have a suggestion regarding this?

**Reply by hugoledoux** [reply]
I don't, besides that you'll have to write your own script I'm afraid. Or lobby esri so that they add a CityJSON output to CityEngine (they have for other products I reckon, but not sure).

**Reply by rhagnus** [reply]
Thanks again for your answer. ArcGISPro is giving json output with adding the z values. I wonder if there is a way to reconstruct the json data as cityjson? Or do I have to write some code for it?

**Reply by hugoledoux** [reply]
I don,t know what flavour of JSON they output, but you surely need to code it (and it might not be that simple).

**Reply by rhagnus** [reply]
Thank you for your previous answers. After a few tries, I think I got the best possible result. I converted the FGDB file extracted from CityEngine to shp. Then, when I converted the shp file to cityjson with FME, I got the result.

![image](https://user-images.githubusercontent.com/57898778/211798542-6060ce7f-ec73-4353-9cc3-0c37916723ef.png)

**Reply by hugoledoux** [reply]
thanks for reporting, this will be useful to others in the future
